Update data_extraction.rst (#452)
CitrineInformatics/citrine-python: docs/source/data_extraction.rst

    table_config = table_configs.register(table_config)
    print("Definition registered as {}".format(table_config.definition_uid))

Registered Table Configs can be built into GEM Tables. For example:

.. code-block:: python

   table = project.tables.build_from_config(table_config)
   project.tables.read(table, "./my_table.csv")
   
The above will build a table, wait for the build job to complete, and save the table locally.
   
However, GEM Tables are sometimes large and time-consuming to build, so the build process can be performed asynchronously with the ``initiate_build`` method.
For example:

.. code-block:: python

    job = project.tables.initiate_build(table_config)

The return type of the ``initiate_build`` method is a :class:`~citrine.resources.job.JobSubmissionResponse` that contains a unique identifier for the submitted job.

> This identifier can be used to get the status of the job via the ``get_job_status`` method, which returns a :class:`~citrine.resources.ara_job.JobStatusResponse`.
> The :class:`~citrine.resources.ara_job.JobStatusResponse` contains a ``status`` string describing the state of the job and an ``output`` map that contains the table id and version.

Reviewer: ```suggestion
```
`get_job_status` is deprecated, as is the entire job status interface.  Short of a feature request to actually allow for user polling of the server, this should be removed from the documentation.  In digging up what would need to get done, at this point we'd either need to craft new methods or have the user roll their own URLs.

    from time import sleep
    # 1. Submit the GEM Table build job
    job = project.tables.initiate_build(table_config)
    # 2. Poll the Job Status every second
    while True:

>         status = project.table_configs.get_job_status(job.job_id)

Reviewer: This code runs but issues warnings.  If correcting this is out of scope for this PR, we don't want it to go disappear off our radar.

Author: What's the warning message? I don't remember seeing one, but I often suppress warnings in my workflows.

Author: Just checked, I don't get one using `initiate_build` (using citrine v.0.61 though)

Author: Still a little confused on this - if it's a deprecated method, we should remove it from the docs. If not, we should make a card to address it. But I can't reproduce the warning you're getting.

Reviewer: ```
<ipython-input-11-6c84f4e184d2>:6: DeprecatedWarning: get_job_status is deprecated. Table build job status does not have a stable format. Please use TableCollection.build_from_config or TableCollection.get_by_build_job
  status = project.table_configs.get_job_status(job.job_id)
```
Issued under both Citrine 0.66 and 0.70.

Author: Oh so it's the part where we query for the job status. Yeah, if you know the correct way to do that that would be great!

Reviewer 2: @kroenlein Did you look into this? I think the example should be something like:

```python
# 1. Submit the GEM Table build job
job = project.tables.initiate_build(table_config)
# 2. Get the table resource as an object
table = project.tables.get_by_build_job(job)
# 3. Download the table
project.tables.read(table, "./my_table.csv")
```

`get_by_build_job` already waits for the job to complete (and checks the status afterwards), so I don't think we don't need explicit polling.

Author: @latture I think it's smart to remove the polling, since the other method waits for completion. In this case, I think it's good to separate the `initiate_build` and `get_by_build_job` methods, since running them back to back will result in an error if you don't wait. 

See the a2d55e8 commit, Line 174-198 for my edits.

It still includes a reference to `get_job_status` in the text. Is there another way to tell if the build job has completed or not?

Reviewer 2: > since running them back to back will result in an error if you don't wait.

Maybe I'm misunderstanding your comment, but why would you get an error if you run them back to back?

> It still includes a reference to get_job_status in the text. Is there another way to tell if the build job has completed or not?

I don't see a dedicated (non-deprecated) method to get the status. @kroenlein Do you know if one? Given a session and job ID you can get it using

```python
path = 'projects/{}/execution/job-status'.format(project_id)
params = {'job_id': job_id}
response = session.get_resource(path=path, params=params)
status: JobStatusResponse = JobStatusResponse.build(response)
```

but we generally try to avoid explicit session calls.

Reviewer: I didn't get a chance to run with it.  I would argue we should probably have async demo code present as well, but the synchronous call example was also in the documentation.  The presumed path would be through the resources.job module in citrine-python.

        if status.status in ['Success', 'Failure']:
            break
        sleep(1)
    if status.status == 'Success':
        # 3. Get the table resource as an object
        table = project.tables.get_by_build_job(job)
        # 4. Download the table
        project.tables.read(table, "./my_table.csv")

> The return type of the ``initiate_build`` method is a :class:`~citrine.resources.ara_job.JobSubmissionResponse` that contains a unique identifier for the submitted job.

Reviewer: ```suggestion
The return type of the ``initiate_build`` method is a :class:`~citrine.resources.job.JobSubmissionResponse` that contains a unique identifier for the submitted job.
```

This identifier can be used to get the status of the job via the ``get_job_status`` method, which returns a :class:`~citrine.resources.ara_job.JobStatusResponse`.
The :class:`~citrine.resources.ara_job.JobStatusResponse` contains a ``status`` string describing the state of the job and an ``output`` map that contains the table id and version.

The table id and version can be used to get a :class:`~citrine.resources.table.Table` resource that provides access the table.

> You can also use the :class:`~citrine.resources.ara_job.JobStatusResponse` to return the :class:`~citrine.resources.table.Table` resource directly with the ``get_by_build_job`` method.

Reviewer: Obsolete reference
```suggestion
You can also use the :class:`~citrine.resources.job.JobStatusResponse` to return the :class:`~citrine.resources.gemtables.GemTable` resource directly with the ``get_by_build_job`` method.
```

  * :class:`~citrine.gemtables.variables.AttributeInOutput`: for when attributes occur both in a process output and one or more of its inputs
  * :class:`~citrine.gemtables.variables.IngredientQuantityByProcessAndName`: for the specific case of the volume fraction, mass fraction, number fraction, or absolute quantity of an ingredient
  * :class:`~citrine.gemtables.variables.IngredientQuantityInOutput`: for the quantity of an ingredient between the terminal material and a given set of processes (useful for ingredients used in multiple processes)

* Identifiers

  * :class:`~citrine.gemtables.variables.RootInfo`: for fields defined on the material at the root of the Material History, like the name of the material
  * :class:`~citrine.gemtables.variables.RootIdentifier`: for the id of the Material History, which can be used as a unique identifier for the rows
  * :class:`~citrine.gemtables.variables.IngredientIdentifierByProcessTemplateAndName`: for the id of the material being used in an ingredient, which can be used as a key for looking up that input material

>   * :class:`~citrine.gemtables.variables.IngredientIdentifierInOutput`: for the id of a material used in an ingredient between the terminal material and a given set of processes (useful for ingredients used in multiple processes)

Reviewer: Lookin' good.

  * :class:`~citrine.gemtables.variables.RootInfo`: for fields defined on the material at the root of the Material History, like the name of the material
  * :class:`~citrine.gemtables.variables.RootIdentifier`: for the id of the Material History, which can be used as a unique identifier for the rows
  * :class:`~citrine.gemtables.variables.IngredientIdentifierByProcessTemplateAndName`: for the id of the material being used in an ingredient, which can be used as a key for looking up that input material

>   * :class:`~citrine.gemtables.variables.IngredientIdentifierInOutput`: for the id the material being used in an ingredient used in multiple processes

Author: I agree with your description. See the latest commit to see the changed language.
I only changed it in two places though (`IngredientIdentifierInOutput` and `IngredientQuantityInOutput`). What's the third you mentioned?
